### PR TITLE
chore: improve intellisense in codebase

### PR DESCRIPTION
**Pull request by makedopamine**
- Compatible with the latest Vue language plugin.  It no longer adds the `GlobalComponents` declaration from @vue/runtime-core to the global scope. See [fda6ad3](https://github.com/vuejs/language-tools/commit/fda6ad360490bb2822ae5309bdd9530ed75cc7a5#diff-1d432278152c845126fa9e323391375cf8fa8806a2cad5c1da6282dae4019947L15)
- Remove a few lines of code I think are redundant.

### CODE REVIEW

**Comment by tolking** [comment]
I noticed that the `components.d.ts` file generated by `unplugin-vue-components` in the docs directory also has a similar issue. Perhaps we should update the version of `unplugin-vue-components`

File: .npmrc [hunk 1]
```diff
@@ -1,4 +1,5 @@
 shamefully-hoist=true
 strict-peer-dependencies=false
 shell-emulator=true
-package-manager-strict=false
\ No newline at end of file
+package-manager-strict=false
+https-proxy=socks://127.0.0.1:1080
```
Comment (tolking) [inline .npmrc:5]: ??
Perhaps this file should not be modified

Comment (makedopamine) [reply .npmrc:5]: ...my bad

**Review by tolking — APPROVED** [review]

File: docs/components.d.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
-// We suggest you to commit this file into source control
+/* eslint-disable */
+// @ts-nocheck
+// Generated by unplugin-vue-components
 // Read more: https://github.com/vuejs/core/pull/3399
-import '@vue/runtime-core'
+export {}
 
-declare module '@vue/runtime-core' {
+/* prettier-ignore */
```
Comment (btea) [inline docs/components.d.ts:7]: This file is included in both the `.prettierignore` and the `.eslintignore` files, so there should be no need to add relevant comments here.

Comment (makedopamine) [reply docs/components.d.ts:7]: True, but the file is auto-generated, we shouldn't modify it manually.

Comment (btea) [reply docs/components.d.ts:7]: Okay.

**Review by btea — APPROVED** [review]

**Pull request merged**